include/StringTools.h

```diff
@@ -0,0 +1,294 @@
+    StringVector MEZZ_LIB Split(const String& Source, const String& Delims = " \t\n", const Whole MaxSplits = 0);
+
+    ///////////////////////////////////////////////////////////////////////////////
+    // Pattern Matching
+
+    /// @brief Checks a String to see if it starts with a specific pattern.
+    /// @param Str The String to check.
+    /// @param Pattern The sequence to check for at the start of the String.
+    /// @param CaseSensitive If true, the pattern will be matched exactly, otherwise differing cases will be matched.
+    /// @return Returns true if the String starts with the provided pattern, false otherwise.
+    Boole MEZZ_LIB StartsWith(const StringView Str, const StringView Pattern, const Boole CaseSensitive);
+    /// @brief Checks a String to see if it ends with a specific pattern.
+    /// @param Str The String to check.
+    /// @param Pattern The sequence to check for at the end of the String.
+    /// @param CaseSensitive If true, the pattern will be matched exactly, otherwise differing cases will be matched.
+    /// @return Returns true if the string ends with the provided pattern, false otherwise.
+    Boole MEZZ_LIB EndsWith(const StringView Str, const StringView Pattern, const Boole CaseSensitive);
+
+    /// @brief Checks to see if a string matches a specific pattern.
+    /// @details This function supports a number of features similar to globbing/wildcard matching implementations
+    /// that exist elsewhere. @n
+    /// A "[" with a matching "]" will encase one or more ranges that can be checked against a single character.
+    /// Ranges defined within square brackets must have a hyphen "-" separating the two extents of the range. They
+    /// must also be the same category of character. For example "[A-Z]" is valid as both characters are upper case,
+    /// but "[A-z]" is not. Upper case, lower case, or numbers can be defined in a range, but cannot be mixed when
+    /// defining one. However multiple ranges can be defined back to back. For example "[A-Za-z]" is valid. Defining
+    /// multiple ranges will only expand the valid matches for a single character though. Defining multiple ranges
+    /// will not cause this to match multiple characters. @n
+    /// A "*" (asterisk) will match 0 or more characters of any value. @n
+    /// A "?" (question mark) will match 0 or 1 character of any value.
+    /// @param Source The string to check.
+    /// @param Pattern The pattern to try and match to the source string.
+    /// @param CaseSensitive Whether or not the pattern should match case exactly.
+    /// @return Returns true if the source string matches the specified pattern.
+    Boole MEZZ_LIB PatternMatch(const StringView Source, const StringView Pattern, const Boole CaseSensitive);
+
+    ///////////////////////////////////////////////////////////////////////////////
+    // Convert-From-String Functions
+
+    /// @brief Converts a String to any.
```

Comment:
I feel like this needs the phrase "Lexicographical Conversion"

Reply:
Since you highlighted multiple parts here....do you mean the section header or the method brief doc?